test/e2e/storage/utils/pod.go

```diff
@@ -69,17 +72,21 @@ func StartPodLogs(f *framework.Framework, driverNamespace *v1.Namespace) func()
 		// keeps each directory name smaller (the full test
 		// name at one point exceeded 256 characters, which was
 		// too much for some filesystems).
-		to.LogPathPrefix = framework.TestContext.ReportDir + "/" +
-			strings.Join(components, "/") + "/"
+		logDir := framework.TestContext.ReportDir + "/" + strings.Join(components, "/")
+		to.LogPathPrefix = logDir + "/"
+
+		err := os.MkdirAll(logDir, 0755)
+		framework.ExpectNoError(err, "create pod log directory")
+		f, err := os.Create(path.Join(logDir, "pod-event.log"))
+		framework.ExpectNoError(err, "create pod events log file")
+		podEventLog = f
 	}
 	podlogs.CopyAllLogs(ctx, cs, ns, to)
 
-	// pod events are something that the framework already collects itself
-	// after a failed test. Logging them live is only useful for interactive
-	// debugging, not when we collect reports.
-	if framework.TestContext.ReportDir == "" {
-		podlogs.WatchPods(ctx, cs, ns, ginkgo.GinkgoWriter)
-	}
+	// The framework doesn't know about the driver pods because of
+	// the separate namespace.  Therefore we always capture the
+	// events ourselves.
+	podlogs.WatchPods(ctx, cs, ns, podEventLog)
```

Comment:
Another improvement for later: extend WatchPods so that it accepts an optional closer and let it close the file when done with it. Currently it assumes that the caller owns the file (like the GinkgoWriter), but when we create it here, we don't know when to close.

Reply:
/hold

This PR is not going to be included in today's periodic job run, so I might as well tune it some more...

Reply:
/hold cancel

Implemented.